### PR TITLE
Enhance transfer coins error messages

### PR DESCRIPTION
**Pull request by aoudiamoncef**
* [x] document all added functions
* [x] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

related to #3464

### CODE REVIEW

File: massa-execution-worker/src/speculative_ledger.rs [hunk 1]
```diff
@@ -179,14 +183,15 @@ impl SpeculativeLedger {
                                 .checked_sub(self.storage_costs_constants.ledger_entry_base_cost)
                                 .ok_or_else(|| {
                                     ExecutionError::RuntimeError(
-                                        "overflow in subtract ledger cost for addr".to_string(),
+                                        format!("overflow in subtract ledger cost {} for new crediting address {}", to_addr, self.storage_costs_constants.ledger_entry_base_cost),
```
Comment (Eitu33) [inline massa-execution-worker/src/speculative_ledger.rs:186]: That's an underflow not an overflow, was already wrong before

File: massa-execution-worker/src/speculative_ledger.rs [hunk 2]
```diff
@@ -179,14 +183,15 @@ impl SpeculativeLedger {
...
-                        return Err(ExecutionError::RuntimeError(
-                            "insufficient amount to create receiver address".to_string(),
-                        ));
+                        return Err(ExecutionError::RuntimeError(format!(
+                            "insufficient amount {} to create crediting address {}",
+                            amount, to_addr
+                        )));
                     }
                 }
                 // if `from_addr` is none and `to_addr` doesn't exist try to create it from coins sent
```
Comment (Eitu33) [inline massa-execution-worker/src/speculative_ledger.rs:197]: Not related to these changes but this comment is completely wrong, here `to_addr` exists

File: massa-execution-worker/src/speculative_ledger.rs [hunk 3]
```diff
@@ -202,7 +207,7 @@ impl SpeculativeLedger {
                                 .checked_sub(self.storage_costs_constants.ledger_entry_base_cost)
                                 .ok_or_else(|| {
                                     ExecutionError::RuntimeError(
-                                        "overflow in subtract ledger cost for addr".to_string(),
+                                        format!("overflow in subtract ledger cost {} for new crediting address {}", to_addr, self.storage_costs_constants.ledger_entry_base_cost),
```
Comment (Eitu33) [inline massa-execution-worker/src/speculative_ledger.rs:210]: This is an underflow as well

File: massa-execution-worker/src/speculative_ledger.rs (hunk 1 above)
Comment (sydhds) [reply massa-execution-worker/src/speculative_ledger.rs:186]: I would keep overflow here as it is more meaningful than underflow ?

File: massa-execution-worker/src/speculative_ledger.rs (hunk 2 above)
Comment (aoudiamoncef) [reply massa-execution-worker/src/speculative_ledger.rs:197]: Oups, I'll fix it

Comment (aoudiamoncef) [reply massa-execution-worker/src/speculative_ledger.rs:197]: image

Correct me if I'm wrong: this means: `to_add` is valid but does't exists in the ledger ?

File: massa-execution-worker/src/speculative_ledger.rs (hunk 3 above)
Comment (aoudiamoncef) [reply massa-execution-worker/src/speculative_ledger.rs]: done

File: massa-execution-worker/src/speculative_ledger.rs (hunk 1 above)
Comment (aoudiamoncef) [reply massa-execution-worker/src/speculative_ledger.rs]: done

File: massa-execution-worker/src/speculative_ledger.rs (hunk 2 above)
Comment (Eitu33) [reply massa-execution-worker/src/speculative_ledger.rs]: Yeah that's it, actually now that I read it again I can understand the comment but it's not very clear still

**Pull request merged**